node/src/types/chainspec/protocol_config.rs

```diff
@@ -29,6 +29,8 @@ pub struct ProtocolConfig {
     /// Any arbitrary updates we might want to make to the global state at the start of the era
     /// specified in the activation point.
     pub(crate) global_state_update: Option<GlobalStateUpdate>,
+    /// The era ID in which the last emergency restart happened.
+    pub(crate) last_emergency_restart: Option<EraId>,
```

Comment:
Could this be represented using `ActivationPoint`? For the no-emergency-restart-yet case that would be a Genesis timestamp? I am wondering whether we can "simplify" it (not that this is terrible complex right now).

Reply:
Hmm - I like the idea of using an `ActivationPoint` rather than an `EraId` as it better represents the meaning, but I'd keep it wrapped in an `Option` for clarity.  I think it'd be confusing to see a chainspec.toml with an entry like `last_emergency_restart = '2021-04-14T09:18:27.712266Z'` but for that to mean there had been no emergency restart.

Reply:
Maybe it just needs a different name in the chainspec, different than `last_emergency_restart`.

Reply:
I'd make it either `Option<EraId>` as it is now, or `ActivationPoint` without an `Option` wrapper. It doesn't make much sense to make it `Option<ActivationPoint>`, if the `Some(Genesis)` variant is never going to be used.

Reply:
For what I need this for it is simplest to have this be an Option<EraId>.  Can we make it an activation point later if that becomes a requirement?

Reply:
Sure, I'm happy to leave as is.